packages/shared/trade/src/transaction-sidebar/transaction-sidebar.tsx

```diff
@@ -104,10 +135,26 @@ export const TransactionSidebar = ({
     }
```

Comment:
Can you consolidate `getTokenSpecificHelpText` and all the logic to get the helptext into a single if / else block? It's hard to read when it's slip up across two parts like this.